npe2/_plugin_manager.py

```diff
@@ -318,8 +317,10 @@ def get_writer(
             elif not ext and len(layer_types) == 1:  # No extension, single layer.
                 ext = next(iter(writer.filename_extensions), "")
                 return writer, path + ext
-            else:
-                raise ValueError
+            # When the list of extensions for the writer doesn't match the
+            # extension in the filename, keep searching.
+
+        # Nothing got found
```

Comment:
this fixes some broken tests I was getting on the napari side. napari's CI misses these at the moment because npe2 isn't installed when the test suite is running.